internal/protoschema/jsonschema/jsonschema.go

```diff
@@ -67,6 +68,30 @@ func WithAdditionalProperties() GeneratorOption {
 	}
 }
 
+// WithStrict sets the generator to require input be pre-normalized.
+//
+// When a JSON value is converted to protobuf, the converter uses the protobuf
+// schema to normalize and validate it further. The default generated schema
+// takes this into account, allowing for implicit default values, aliases, and
+// other leniencies.
+//
+// When strict is enabled, the generated schema will not allow these leniencies.
+// Specifically, the JSON schema:
+//   - Requires implicit default values be explicitly set.
```

Comment:
Changes look good but I'm wondering if requiring implicit default values should perhaps be its own option. All of the other changes mentioned below make sense in the context of a `strict` option, but this one is unclear to me.

Reply:
I actually worry about not allowing strings for integers. I decided to make it as strict as possible, so that we can relax and split out options if issues come up.